tools/proguard5.3.2/test/proguard/util/ArrayUtilTest.java

```diff
@@ -0,0 +1,614 @@
+				new short[]{0, 1, 2, 3, 4}, 3,
+				new short[]{0, 1, 2}, 3)
+		);
+		Assert.assertEquals(1, ArrayUtil.compare(
+				new short[]{2, 1, 0}, 3,
+				new short[]{0, 1, 2}, 3)
+		);
+	}
+
+	@Test
+	public void testCompareInt() {
+		Assert.assertEquals(-1, ArrayUtil.compare(
+				new int[]{1, 2}, 2,
+				new int[]{1, 2, 3, 0, 0}, 5)
+		);
+		Assert.assertEquals(-1, ArrayUtil.compare(
+				new int[]{0, 1, 2}, 3,
+				new int[]{2, 1, 0}, 3)
+		);
+		Assert.assertEquals(0, ArrayUtil.compare(
+				new int[0], 0,
+				new int[0], 0)
+		);
+		Assert.assertEquals(0, ArrayUtil.compare(
+				new int[]{0, 1, 2}, 3,
+				new int[]{0, 1, 2}, 3)
+		);
+		Assert.assertEquals(0, ArrayUtil.compare(
+				new int[]{0, 1, 2}, 3,
+				new int[]{0, 1, 2, 3, 4}, 3)
+		);
+		Assert.assertEquals(0, ArrayUtil.compare(
+				new int[]{0, 1, 2, 3, 4}, 3,
+				new int[]{0, 1, 2}, 3)
+		);
+		Assert.assertEquals(1, ArrayUtil.compare(
+				new int[]{2, 1, 0}, 3,
+				new int[]{0, 1, 2}, 3)
+		);
+	}
+
+	@Test
+	public void testCompareComparable() {
+		Assert.assertEquals(-1, ArrayUtil.compare(
+				new Comparable[]{1, 2}, 2,
+				new Comparable[]{1, 2, 3, 0, 0}, 5)
+		);
+		Assert.assertEquals(0, ArrayUtil.compare(
+				new Comparable[0], 0,
+				new Comparable[0], 0)
+		);
+		Assert.assertEquals(0, ArrayUtil.compare(
+				new Comparable[]{0, 1, 2}, 3,
+				new Comparable[]{0, 1, 2}, 3)
+		);
+		Assert.assertEquals(0, ArrayUtil.compare(
+				new Comparable[]{0, 1, 2}, 3,
+				new Comparable[]{0, 1, 2, 3, 4}, 3)
+		);
+		Assert.assertEquals(0, ArrayUtil.compare(
+				new Comparable[]{0, 1, 2, 3, 4}, 3,
+				new Comparable[]{0, 1, 2}, 3)
+		);
+		Assert.assertEquals(1, ArrayUtil.compare(
+				new Comparable[]{2, 1, 0}, 3,
+				new Comparable[]{0, 1, 2}, 3)
+		);
+	}
+
+	@Test
+	public void testExtendArrayBoolean() {
+		Assert.assertArrayEquals(new boolean[0],
+				ArrayUtil.extendArray(new boolean[0], 0)
+		);
+		Assert.assertArrayEquals(new boolean[]{true, true, true, false, false},
+				ArrayUtil.extendArray(new boolean[]{true, true, true}, 5)
+		);
+	}
+
+	@Test
+	public void testEnsureArraySizeBoolean() {
+		boolean[] booleans = new boolean[]{false, false, true};
+
+		Assert.assertArrayEquals(new boolean[]{true, true, true, true},
+				ArrayUtil.ensureArraySize(booleans, 4, true)
+		);
+
+		booleans = new boolean[]{false, false, true, true};
+
+		Assert.assertArrayEquals(new boolean[]{false, false, false, true},
+				ArrayUtil.ensureArraySize(booleans, 3, false)
+		);
+		Assert.assertArrayEquals(new boolean[]{false, false, false, false},
+				ArrayUtil.ensureArraySize(booleans, 4, false)
+		);
+	}
+
+	@Test
+	public void testAddByte() {
+		Assert.assertArrayEquals(new byte[]{0, 0, 0},
+				ArrayUtil.add(new byte[]{1, 0, 0}, 0, (byte) 0)
+		);
+		Assert.assertArrayEquals(new byte[]{1, 0, 0},
+				ArrayUtil.add(new byte[]{1, 0, 0}, 1, (byte) 0)
+		);
+		Assert.assertArrayEquals(new byte[]{1, 0, 0, 0},
+				ArrayUtil.add(new byte[]{1, 0, 0}, 3, (byte) 0)
+		);
```

Comment:
Try to use different values to show that the byte value has changed.
Ie {1,1,1} when changed {1,0,1}

Reply:
This was just a test to show what happens when you add a byte on the end and so the rest was not changed

Reply:
Yes.. but since the byte added is the same as the bytes in the array.. its a bit difficult to notice